azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
I think it's reasonable limitation. Anyway, "backing up" to Boost for standalone ASIO may be added later if requested by users.

I can take a task to switch threading related stuff: std will be used by default unless `AZMQ_DISABLE_STD_THREAD` is defined.